### ISSUE TITLE
Missing whitespace when copying codeblock contents

### ISSUE THREAD

**Issue by alreadypro** [vBread/bd-contributions#29]
## Description

New lines & (some) tabs are not copied to clipboard when a codeblock's content is copied.

This could be fixed multiple ways, but a favorable solution would be to add a "copy code" button on the codeblock.

### Reproduction Steps

1) Select all of the code in a codeblock
2) Ctrl + C to copy it
3) Ctrl + V in your favorite editor
4) ???
5) where are my newlines & (some) tabs?????

- Plugin Name: BetterCodeblocks
- Discord Release: Stable
- Platform: Windows 11

**Comment by Setloth** [comment]
![image](https://user-images.githubusercontent.com/44213317/160211971-f3d1f2bf-ab49-4c32-a5f3-61d4f18e93ea.png)
![image](https://user-images.githubusercontent.com/44213317/160211973-585c2667-25ca-42ad-8042-ff6e26dda3e8.png)
also fitting with this, tabs just vanish, might be related

**Comment by justanothercell** [comment]
this is temporarily fixable by editing the source. replace this in line 316:
```js
// comment this out:
// return !min ? content : content.replace(new RegExp(`^ {${min}}`, "gm"), '');
// put this there:
return content
```
idk whether thatbreaks things elsewhere, but for now its better than the two missing whitespaces

**Comment by alreadypro** [comment]
> this is temporarily fixable by editing the source. replace this in line 316:
> 
> ```js
> // comment this out:
> // return !min ? content : content.replace(new RegExp(`^ {${min}}`, "gm"), '');
> // put this there:
> return content
> ```idk whether thatbreaks things elsewhere, but for now its better than the two missing whitespaces
> ```

perhaps you should make a PR?

**Comment by justanothercell** [comment]
nah, im just removing functionality cheaply without knowing what else it might break, there has to be a reason why thats there, and im just basically skipping it. until i know why that is there in the first place that would be a bad idea

**Comment by Setloth** [comment]
![image](https://user-images.githubusercontent.com/44213317/163663017-40f93303-585c-4f66-8a0c-3f0334a89069.png)
me  running some tests console.log'ing the info. the first CONTENT is the variable as it exists in the parameter CONTENT 2 is after the first regex edit, the MIN is the value of `min` and then END is what it returns

This is an explanation of the first regex:
![image](https://user-images.githubusercontent.com/44213317/163663047-cb139c0e-46ec-4f58-a395-4fd05d4a63aa.png)

Second:
![image](https://user-images.githubusercontent.com/44213317/163663056-f8521826-fb9f-4468-b760-a046287ec41d.png)
However, the line this regex is on also will reduce the content as an array by the regex into a number by using `Math.min()` on the values based on the array. Basically, this will compare each value (which represents the index of the returned regex match in the string) to the length of the array, and return the lower set of the numbers. Starting with Infinity, it will then add these numbers together for the sum, and set the `min` variable to it

Then, the last regex, uses the `min` variable and makes a regex
![image](https://user-images.githubusercontent.com/44213317/163663166-628ac3a0-1fb1-4507-a0bf-a152df211bb2.png)
Then, on this line it will check first if min is 0, if it is, it returns the content variable as before, if it isn't, then it will replace content based on the regex above with a blank string `''`

Essentially, the tl;dr, being the only reason that this (stupid) method exists, is to cut out the first line of indents in a code block. AKA, its basically useless. It is very possible that when making the plugin, there was some much that added extra spacing as to mess up the formatting, but currently there is no such issue and it makes no sense to include this. However, I am not the developer for this plugin so I'll leave that up to the developer.


Also, In my copy of the code I am setting the method to:
```js
dedent(content) {
	return content;
}```

As the first regex will remove any tab characters, and that is also pointless.

**Comment by CandiceJoy** [comment]
This is fixed in the latest commit.  I've submitted a pullrequest to make the latest commit load shiki correctly which fixes both bugs for me.
